### DISCUSSION TITLE
Unable to change nextcloud datadir on Windows

### DISCUSSION THREAD

**Discussion by Cailiil2009** [nextcloud/all-in-one#600] [Questions] [answered]
Hi all.
I try to map a Windows 10 directory to nextcloud as datadir. I use the following command to create the container.
 
docker run -it ^
--name nextcloud-aio-mastercontainer ^
--restart always ^
-p 80:80 ^
-p 8080:8080 ^
-p 8443:8443 ^
--volume nextcloud_aio_mastercontainer:/mnt/docker-aio-config ^
--volume //var/run/docker.sock:/var/run/docker.sock:ro ^
-e NEXTCLOUD_DATADIR="/host_mnt/x/nextcloud" ^
nextcloud/all-in-one:latest

Somehow it fails all the time. When I take away the -e part it works perfectly with the standard Data directory.
Can someone help me?

**Comment by szaimen** [comment]
Hello, in theory we support setting these paths but in reality there is most likely a bug in the Windows Docker API. I already tried to debug this and added my findings in this ticket: https://github.com/nextcloud/all-in-one/issues/396

This needs to be reported to the win-docker repository but I didn't and don't have time for that. 

Maybe you can take care of it?

**Reply by Cailiil2009** [reply]
I am a beginner with docker. Would it be possible for me to download the image and change the configuration so it fits my system? Would that even work?

**Reply by szaimen** [reply]
> I am a beginner with docker. Would it be possible for me to download the image and change the configuration so it fits my system? Would that even work?

No, unfortunately there is no way that I am aware of that would make it work. It needs to be reported to https://github.com/docker/for-win

**Reply by szaimen** [reply]
However I will add it to the documentation for now.

**Reply by szaimen** [reply]
documented in https://github.com/nextcloud/all-in-one/commit/36797305d8c29974abcc80733dd7a1b04c05bb99#diff-5a831ea67cf5cf8703b0de46901ab25bd191f56b320053be9332d9a3b0d01d15

**Reply by Cailiil2009** [reply]
Thank you

**Reply by Cailiil2009** [reply]
I probably use A ubuntu in wsl and install nextcloud aio there. That should solve my issue.

**Reply by szaimen** [reply]
Good luck!

**Comment by szaimen** [comment]
Hello, just wanted to inform that I investigated this a bit and it seems like there is actually a way to make it work on Windows. It will need some additional steps but it is possible. See https://github.com/nextcloud/all-in-one/issues/688

**Comment by szaimen** [comment, marked as answer]
Hello, this is now released with 1.3.0 Beta. Feel free to try it out and report back! Thank you! :)
See https://github.com/nextcloud/all-in-one#how-to-switch-the-channel
